### PR TITLE
GUVNOR-3106: [Guided Decision Table] Rule names are not added to index

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3106

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/GuidedDecisionTableModelIndexVisitor.java [hunk 1]
```diff
@@ -54,83 +54,95 @@
...
+        for (List<DTCellValue52> row : model.getData()) {
+            final String ruleName = getPackagePrefix() + "Row " + row.get(0).getNumericValue().longValue() + " " + model.getTableName();
+            addResource(ruleName,
+                        ResourceType.RULE);
+        }
+        //Add parent rule name
         final String parentRuleName = model.getParentName();
-        for ( List<DTCellValue52> row : model.getData() ) {
-            final String ruleName = "Row " + row.get( 0 ).getNumericValue().longValue() + " " + model.getTableName();
-            addResourceReference(ruleName, ResourceType.RULE);
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/GuidedDecisionTableModelIndexVisitor.java]: This was the problem. We were adding a "reference to a rule" not a "rule" itself.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/GuidedDecisionTableModelIndexVisitor.java [hunk 2]
```diff
@@ -54,83 +54,95 @@
...
         //Add Types and Fields used by Actions
-        for ( ActionCol52 c : o.getActionCols() ) {
-            visit( c );
+        for (ActionCol52 c : o.getActionCols()) {
+            visit(c);
         }
         //Add rule names
+        for (List<DTCellValue52> row : model.getData()) {
+            final String ruleName = getPackagePrefix() + "Row " + row.get(0).getNumericValue().longValue() + " " + model.getTableName();
+            addResource(ruleName,
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/GuidedDecisionTableModelIndexVisitor.java]: Now we correctly add a "rule" itself.

**Comment by jomarko** [comment]
@ederign please be aware of the fact, that every PR for guided decision tables should be approved also from QE site. This agreement emerged recently between me and @manstis. I think he also sent some email about this. In future, I would like to ask you to wait with merging until QE approve the PR that relates to GDT. Thanks

**Comment by manstis** [comment]
@jomarko To be honest I only advised @Rikkola and @karreiro as they're the two who usually review GDT related PRs so @ederign can perhaps be excused this time ;-) We're all part of the same family; let's not have a feud!

**Comment by jomarko** [comment]
@manstis I understand, I am sorry if my comment sounded too much offensive. I completely understand that @ederign didn't know about the agreement. I am not offended because of merging without my ack or something. I just want to tell him that some agreement exists. All, sorry for inconvenience.

**Pull request merged**